### PR TITLE
added-spring-boot-security

### PR DESCRIPTION
**Pull request by margosalii**

### CODE REVIEW

File: src/main/java/mate/academy/store/model/User.java [hunk 1]
```diff
@@ -38,4 +47,47 @@ public class User {
 
     @Column(nullable = false)
     private boolean isDeleted = false;
+
+    @ManyToMany(fetch = FetchType.EAGER)
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/store/model/User.java]:
```suggestion
    @ManyToMany
```

use `@EntityGraph(attributePaths = "roles")` in repo instead

File: src/main/java/mate/academy/store/security/UserDetailsServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+@RequiredArgsConstructor
+@Service
+public class UserDetailsServiceImpl implements UserDetailsService {
+    private final UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        return userRepository.findByEmail(email)
+            .orElseThrow(() -> new UsernameNotFoundException("Can't find user by email: " + email));
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/store/security/UserDetailsServiceImpl.java]:
```suggestion
            .orElseThrow(() -> new EntityNotFoundException("Can't find user by email: " + email));
```

File: src/main/java/mate/academy/store/security/UserDetailsServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class UserDetailsServiceImpl implements UserDetailsService {
+    private final UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/store/security/UserDetailsServiceImpl.java]:
```suggestion
    public UserDetails loadUserByUsername(String email) {
```

File: src/main/java/mate/academy/store/service/impl/UserServiceImpl.java [hunk 4]
```diff
@@ -25,7 +30,13 @@ public UserResponseDto register(UserRegistrationRequestDto requestDto)
...
                 + requestDto.getEmail());
         }
-        User user = userMapper.toModel(requestDto);
+        User user = new User();
+        user.setEmail(requestDto.getEmail());
+        user.setPassword(passwordEncoder.encode(requestDto.getPassword()));
+        user.setFirstName(requestDto.getFirstName());
+        user.setLastName(requestDto.getLastName());
+        user.setShippingAddress(requestDto.getShippingAddress());
+        user.getRoles().add(roleRepository.findByName(RoleName.USER));
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/store/service/impl/UserServiceImpl.java]:
```suggestion
        User user = userMapper. toModel(requestDto);
        user.setRoles(Set.of(roleRepository.findByName(RoleName.USER)));
        user.setPassword(encoder.encode(requestDto.getPassword()));
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job. Let's improve it

File: src/main/java/mate/academy/store/model/User.java (hunk 1 above)
Comment (MentorMateAcademy) [reply src/main/java/mate/academy/store/model/User.java]: Not fixed

**Review by MentorMateAcademy — CHANGES_REQUESTED** [review]

**Review by Elena-Bruyako — APPROVED** [review]
well done

**Pull request merged**